eclair-core/src/main/scala/fr/acinq/eclair/transactions/CommitmentSpec.scala

```diff
@@ -33,8 +33,21 @@ sealed trait CommitmentOutput
 object CommitmentOutput {
   case object ToLocal extends CommitmentOutput
   case object ToRemote extends CommitmentOutput
+  case class InHtlc(incomingHtlc: IncomingHtlc) extends CommitmentOutput
```

Comment:
If we only keep one `Htlc(htlc: DirectedHtlc) extends CommitmentOutput` the pattern matching that annoyed us yesterday doesn't work properly?

Reply:
Or maybe `Htlc[T <: DirectedHtlc](htlc: T)`?

Reply:
> Or maybe Htlc[T <: DirectedHtlc](htlc: T)?

I had that in mind but what @araspitzu went with is simpler and will cause less typing headaches imo.

Reply:
Alright let's keep it like that then, it's true that it's simpler.